samples/crypto/spake2p/src/main.c

```diff
@@ -0,0 +1,282 @@
+
+	psa_pake_cs_set_algorithm(&cipher_suite, psa_alg);
+	psa_pake_cs_set_primitive(&cipher_suite, PSA_PAKE_PRIMITIVE(PSA_PAKE_PRIMITIVE_TYPE_ECC,
+								    PSA_ECC_FAMILY_SECP_R1, 256));
+	psa_pake_cs_set_key_confirmation(&cipher_suite, PSA_PAKE_CONFIRMED_KEY);
+
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+	psa_set_key_usage_flags(&key_attributes, PSA_KEY_USAGE_DERIVE);
+	psa_set_key_algorithm(&key_attributes, psa_alg);
+	psa_set_key_type(&key_attributes, PSA_KEY_TYPE_SPAKE2P_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+	psa_set_key_bits(&key_attributes, 256);
+
+	psa_key_id_t key;
+
+	status = psa_import_key(&key_attributes, key_pair, sizeof(key_pair), &key);
+	if (status != PSA_SUCCESS) {
+		LOG_INF("psa_import_key failed. (Error: %d)", status);
+		goto error;
+	}
+
+	psa_set_key_usage_flags(&key_attributes, PSA_KEY_USAGE_DERIVE);
+	psa_set_key_algorithm(&key_attributes, psa_alg);
+	psa_set_key_type(&key_attributes, PSA_KEY_TYPE_SPAKE2P_PUBLIC_KEY(PSA_ECC_FAMILY_SECP_R1));
+	psa_set_key_bits(&key_attributes, 256);
+
+	psa_key_id_t key_server;
+
+	status = psa_import_key(&key_attributes, public_key, sizeof(public_key), &key_server);
+	if (status != PSA_SUCCESS) {
+		LOG_INF("psa_import_key failed. (Error: %d)", status);
+		goto error;
+	}
+
+	psa_reset_key_attributes(&key_attributes);
+
+	psa_pake_operation_t client_op = PSA_PAKE_OPERATION_INIT;
+	/* Client side pake setup. */
+	{
+		status = psa_pake_setup(&client_op, key, &cipher_suite);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_setup failed. (Error: %d)", status);
+			goto error;
+		}
+
+		status = psa_pake_set_role(&client_op, PSA_PAKE_ROLE_CLIENT);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_set_role failed. (Error: %d)", status);
+			goto error;
+		}
+
+		status = psa_pake_set_user(&client_op, "client", strlen("client"));
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_set_user failed. (Error: %d)", status);
+			goto error;
+		}
+
+		status = psa_pake_set_peer(&client_op, "server", strlen("server"));
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_set_peer failed. (Error: %d) 123", status);
+			goto error;
+		}
+	}
+
+	psa_pake_operation_t server_op = PSA_PAKE_OPERATION_INIT;
+	/* Server side pake setup. */
+	{
+		status = psa_pake_setup(&server_op, key_server, &cipher_suite);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_setup failed. (Error: %d)", status);
+			goto error;
+		}
+
+		status = psa_pake_set_role(&server_op, PSA_PAKE_ROLE_SERVER);
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_set_role failed. (Error: %d)", status);
+			goto error;
+		}
+
+		status = psa_pake_set_peer(&server_op, "client", strlen("client"));
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_set_peer failed. (Error: %d)", status);
+			goto error;
+		}
+
+		status = psa_pake_set_user(&server_op, "server", strlen("server"));
+		if (status != PSA_SUCCESS) {
+			LOG_INF("psa_pake_set_user failed. (Error: %d)", status);
+			goto error;
+		}
+	}
```

Comment:
In the ecjpake sample we have this in `pake_setup` function might be nice here as well

Reply:
I can do that